### ISSUE TITLE
Add prop to disable line numbers on code blocks

### ISSUE THREAD

**Issue by zachkirsch** [fern-api/fern#4450]
It would be awesome if I could conditionally disable the line numbers on code blocks

<img width="231" alt="Screenshot 2024-08-28 at 5 49 09 PM" src="https://github.com/user-attachments/assets/8655cc02-2d7c-4536-8dfd-dda83b6482d9">

**Comment by cwcowell** [comment]
👍 to this. Sometimes a page includes multiple snippets, which together form a single block of code, so it’s potentially confusing to readers to have each block restart the numbering at 1.

Even better would be to be able to **either** disable **or** specify a starting line number for a code block.

**Comment by hkosova** [comment]
It's possible to disable line numbers in _all_ code blocks with a bit of CSS:

```css
.code-block-line-gutter {
    display: none;
}
```

**Comment by cwcowell** [comment]
Building on @hkosova ’s idea, you can hide line numbers for particular code blocks like this:

In a custom CSS file:

```css
.no-line-numbers .code-block-line-gutter {
    display: none;
}
```

In your MDX file:

```html
<div class="no-line-numbers">

your code block goes here

</div>
```

**Comment by cwcowell** [comment]
Although… I just realized that both @hkosova ’s approach and mine remove any vertical whitespace in your code block for some reason, so there’s that downside:

So this in your Markdown:

<img width="287" alt="image" src="https://github.com/user-attachments/assets/94d300eb-965d-4166-a7e1-1b8756e9f5e5">


Renders like this, with the space removed:

<img width="207" alt="image" src="https://github.com/user-attachments/assets/aa4e49d8-582d-4ee8-ae3f-60b43221f41e">


TL;DR: we still need this feature from Fern!

**Comment by hkosova** [comment]
>  I just realized that both @hkosova ’s approach and mine remove any vertical whitespace in your code block for some reason

@cwcowell try this version:

```css
.code-block-line-gutter > span {
    display: none;
}
.code-block-line-content > .line {
    display: inline-block;
}
```

**Comment by cwcowell** [comment]
@hkosova Brilliant! That works perfectly.

For reference, here’s a snippet from my `fern/docs/assets/styles.css` using your solution:

```css
.no-line-numbers .code-block-line-gutter > span {
    display: none;
}

.no-line-numbers .code-block-line-content > .line {
    display: inline-block;
}
```

And here’s a snippet from an `.mdx` file that uses the new CSS classes:

<img width="324" alt="image" src="https://github.com/user-attachments/assets/1c49df0f-84d4-4771-9dea-507bf09cc656">

And here’s the rendered result with the newline intact:

<img width="340" alt="image" src="https://github.com/user-attachments/assets/a3c1407c-92a5-4d60-840e-0d6486af0ed0">

**Comment by dannysheridan** [comment]
Love this! I'm working (hard) on adding another Frontend Engineer to our team so that we can turn around reported issues like this faster. I appreciate the cross-collaboration!

**Comment by davidkonigsberg** [comment]
@chdeskur is this still on the roadmap?
